controllers/utils.go

```diff
@@ -470,3 +471,46 @@ func addRestoreLabelSelector(
 		)
 	}
 }
+
+// update BackupSchedule status and remove velero schedules
+// when the BackupSchedule is paused or in BackupCollision
+func updateBackupSchedulePhaseWhenPaused(
+	ctx context.Context,
+	c client.Client,
+	veleroScheduleList veleroapi.ScheduleList,
+	backupSchedule *v1beta1.BackupSchedule,
+	phase v1beta1.SchedulePhase,
+	msg string,
+) error {
+	scheduleLogger := log.FromContext(ctx)
+
+	if phase != v1beta1.SchedulePhasePaused && phase != v1beta1.SchedulePhaseBackupCollision {
+		// do not process any other type of schedule here
+		return nil
+	}
+
+	//update status
+	backupSchedule.Status.Phase = phase
+	backupSchedule.Status.LastMessage = msg
+
+	backupSchedule.Status.VeleroScheduleCredentials = nil
+	backupSchedule.Status.VeleroScheduleManagedClusters = nil
+	backupSchedule.Status.VeleroScheduleResources = nil
+
+	err := c.Status().Update(ctx, backupSchedule)
```

Comment:
Should we do the status update only after all the velero schedules have been successfully deleted? (just after the delete over veleroScheduleList.Items below) That way you know when you see the status is paused, we're done pausing.

Reply:
I moved the BackupSchedule update after the schedules deletion; the code is being called only after the schedules are being successfully deleted.